distribution/ddf-common/src/main/resources/security/default.policy

```diff
@@ -31,153 +31,340 @@ deny {
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}users.properties", "read,write";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}users.attributes", "read,write";
+}
+
+grant codeBase "file:/admin-core-jolokia/javax.servlet-api/org.apache.shiro.core/org.eclipse.jetty.io/org.eclipse.jetty.security/org.eclipse.jetty.server/org.eclipse.jetty.servlet/org.ops4j.pax.web.pax-web-extender-whiteboard/org.ops4j.pax.web.pax-web-jetty/org.ops4j.pax.web.pax-web-runtime/org.apache.cxf.cxf-rt-transports-http/org.eclipse.jetty.servlet/org.eclipse.jetty.util/platform-filter-clientinfo/platform-filter-delegate/platform-filter-response/security-certificate-generator/security-filter-authorization/security-filter-login/security-filter-web-sso" {
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}keystores${/}-", "read,write";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}system.properties", "read,write";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}users.properties", "read,write";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}users.attributes", "read,write";
+    permission java.security.SecurityPermission "insertProvider";
+}
+
+grant codeBase "file:/security-core-api/security-core-impl/security-idp-server/security-idp-client/security-idp-plugin-subjectconfirmation" {
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}ws-security${/}server${/}-", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}ws-security${/}issuer${/}-", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}keystores${/}-", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}metadata", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}metadata${/}-", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}users.properties", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}org.ops4j.pax.web.cfg", "read";
+}
+
+grant codeBase "file:/org.apache.cxf.cxf-rt-transports-http/org.ops4j.pax.web.pax-web-runtime/org.ops4j.pax.web.pax-web-jetty/org.ops4j.pax.web.pax-web-extender-whiteboard/org.ops4j.pax.web.pax-web-extender-war/org.eclipse.jetty.util/org.eclipse.jetty.server/org.eclipse.jetty.servlet/org.eclipse.jetty.security/org.eclipse.jetty.io/javax.servlet-api/org.apache.shiro.core/security-filter-web-sso/security-filter-login/security-filter-authorization/platform-filter-response/platform-filter-clientinfo/platform-filter-delegate/security-interceptor-guest-wrapper/security-interceptor-guest/security-handler-api" {
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}metadata", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}metadata${/}-", "read";
+    permission java.io.FilePermission "${ddf.home.perm}etc${/}keystores${/}-", "read";
+    permission java.io.FilePermission "etc${/}keystores${/}-", "read";
```

Comment:
❗️ Missing root of `etc`.

Reply:
It needs to match exactly this permission or it doesn't work. There are perms for both with and without the root.

Reply:
Might want to leave a comment to that effect. Oy. :rofl: